lib/stats/stats-cluster.c

```diff
@@ -25,6 +25,104 @@
 
 #include <string.h>
 
+typedef struct _StatsComponentsElement
+{
+  GString *name;
+  gint index;
+} StatsComponentsElement;
+
+static GHashTable *stats_components = NULL;
+static StatsComponentsElement *mapping_from_component_number_to_element[SCS_SOURCE_MASK];
+
+static guint
+_stats_components_element_hash_cb(gconstpointer element)
+{
+  if (element)
+    return g_str_hash(((StatsComponentsElement *)element)->name->str);
+
+  return 0;
+}
+
+static gboolean
+_stats_components_names_are_equal_cb(gconstpointer a, gconstpointer b)
+{
+  return g_strcmp0(((StatsComponentsElement *)a)->name->str, ((StatsComponentsElement *)b)->name->str) == 0;
+}
+
+static void
+_stats_components_destroy_element_cb(gpointer e)
+{
+  StatsComponentsElement *element = (StatsComponentsElement *)e;
+  if (element)
+    {
+      g_string_free(element->name, TRUE);
+      g_free(element);
+    }
+}
+void
+stats_components_init()
+{
+  gint i;
+
+  if (!stats_components)
+    stats_components = g_hash_table_new_full(_stats_components_element_hash_cb,
+                                             _stats_components_names_are_equal_cb,
+                                             NULL,
+                                             _stats_components_destroy_element_cb);
+
+  for (i = 0; i < SCS_SOURCE_MASK; i++)
+    {
+      mapping_from_component_number_to_element[i] = NULL;
+    }
+}
+
+void
+stats_components_deinit()
+{
+  gint i;
+  g_hash_table_destroy(stats_components);
+  stats_components = NULL;
+
+  for (i = 0; i < SCS_SOURCE_MASK; i++)
+    mapping_from_component_number_to_element[i] = NULL;
+}
+
+int
+stats_components_get_component_index(const gchar *name)
```

Comment:
This function does more than its name implies. It would be nice, if it would be renamed or the part of its functionality would be extracted into a separate function.